### PR TITLE
Local setup taking precedence over Docker

### PR DESCRIPTION
**Pull request by jmolina4**
We have seen that running the integration tests in docker takes a long time impacting the experience when doing TDD in the interview. We want to change that so that Local setup is the preferred way to go.

### CODE REVIEW

**Review by lleites — APPROVED** [review]

**Review by daialv — APPROVED** [review]

File: README-DOCKER.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+```bash
+JOB=citibike_ingest ./go.sh run-colima-job
+```
+
+#### Distance calculation
+This job takes bike trip information and calculates the "as the crow flies" distance traveled for each trip.
+It reads the previously ingested data parquet files.
+
+Hint:
+ - For distance calculation, consider using [**Harvesine formula**](https://en.wikipedia.org/wiki/Haversine_formula) as an option.
```
Comment (daialv) [inline README-DOCKER.md:191]: This typo has been bugging me for so long (almost 2 years). @jmolina4 can you please change Ha_r_vesine into Have_r_sine? Thanks!

Comment (daialv) [reply README-DOCKER.md:191]: Could you please take a look at README.md on this as well?

**Review by daialv — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -3,91 +3,44 @@ This is a collection of _Python_ jobs that are supposed to transform data.
...
-Full list of commands for Mac and Linux users is as follows: 
-
-| S.No.      | Command | Action     |
-| :---:        |    :----   |          :--- |
-| 1      | ./go.sh lint       | Static analysis, code style, etc. (please install poetry if you would like to use this command)   |
-| 2      | ./go.sh linting       | Static analysis, code style, etc. (please install poetry if you would like to use this command)   |
-| 3      | ./go.sh install-with-docker-desktop       | Install the application requirements along with docker desktop   |
-| 4      | ./go.sh install-with-colima       | Install the application requirements along with colima   |
-| 5      | ./go.sh start-colima       | Start Colima   |
-| 6      | ./go.sh run-local-unit-test       | Run unit tests on local machine   |
```
Comment (svishal9) [inline README.md:65]: What are we trying to achieve by moving back to make file syntax? AFAIR we moved to single "go" file to ensure uniformity across different platforms (Mac, Linux, Win, Container).
How moving to make file resolve performance issue?

File: README.md [hunk 3]
```diff
@@ -3,91 +3,44 @@ This is a collection of _Python_ jobs that are supposed to transform data.
 These jobs are using _PySpark_ to process larger volumes of data and are supposed to run on a _Spark_ cluster (via `spark-submit`).
 
 ## Pre-requisites
-
-We use [`batect`](https://batect.dev/) to dockerise the tasks in this exercise.
```
Comment (svishal9) [inline README.md:7]: What is the issue with container which impacts performance? Is download of image taking time? is container runtime causing issue? is the issue with spark driver? what if we encounter this issue in local as well?
I feel we should investigate.

**Review by svishal9 — CHANGES_REQUESTED** [review]
1. I'm not convinced with this change because it does not resolve the underlying problem.
2. moving back to make file pattern will surface all those issues because of which we moved to "go" file pattern. it will also complicate as way of execution will change across platforms.
3. have we tested the instructions on Windows?

File: README.md (hunk 3 above)
Comment (jmolina4) [reply README.md:7]: The issue is when running integration tests.

We discussed that and yes, we need to investigate a way to use docker efficiently, but in the meantime with need to have a setup where we can do tdd to assess the candidate skills.

File: README.md (hunk 2 above)
Comment (svishal9) [reply README.md:65]: Thanks for the response @jmolina4 !
this comment was more about changing the patter back to makefile. How is it going to help?
All our repos (python/java/scala) use "go" file pattern.

File: README.md (hunk 3 above)
Comment (svishal9) [reply README.md:7]: Hi @jmolina4 ,

Following is the result from my machine:

image

I'm not able to simulate the issue. Hence the suggestion of finding the root cause.

On a different note, even if the issue exists, this change is not going to fix it. So if someone has to use container, they would not be able to get better experience.

**Comment by svishal9** [comment]
> 1. I'm not convinced with this change because it does not resolve the underlying problem.
> 2. moving back to make file pattern will surface all those issues because of which we moved to "go" file pattern. it will also complicate as way of execution will change across platforms.
> 3. have we tested the instructions on Windows?

any comments on Windows testing @jmolina4 ?

File: README.md (hunk 2 above)
Comment (jmolina4) [reply README.md:65]: I don't think this is a final solution, but instead a temporary one. The idea would be to go back to container approach if we find a way to speed it up.

Honestly I just changed the order of the READMEs, so I didn't do any change to the content to the existing readmes.

**Comment by jmolina4** [comment]
> any comments on Windows testing @jmolina4 ?

I saw @daialv will give it a try in a windows machine.

File: README.md (hunk 2 above)
Comment (jmolina4) [reply README.md:65]: I just replaced the the make references to use go.sh or go.ps

**Pull request closed without merging**